src/renderer/utils/markdown.ts

```diff
@@ -130,6 +130,60 @@ const Markdown = {
 
     },
 
+    // Wikilink
+    wikilink () {
+
+      const {path: notesPath} = Config.notes;
+
+      if ( !notesPath ) return [];
+
+      var matches = [];
+      return [
+        { // Link
+          type: 'lang',
+          regex: /(?<!```(?: *)(?:[^\s`~]*)\n(?:[\s\S]*?)|`)\[{2}(.*?\|?.*?)\]{2}/g,
```

Comment:
This regex is too complicated, we shouldn't use any regexes that are too difficult to reason about. Can you explain what it's supposed to do?

Reply:
Btw from trying to understand it I think it is too brittle to be used:

- I don't think this thing supports things like `<b>```</b>`, which is perfectly valid
- It also doesn't support backticks inside code blocks 
- I also don't understand what `~` has anything to do with this
- `\[{2}` is more complicated to read than `\[\[` IMHO 
- `(.*?\|?.*?)` this part seems nonsensical to me, what it's supposed to match exactly? I don't think it will handle multi-word titles such as "Foo Bar", unless an explicit description has been provided.

Reply:
This is part `(?<!```(?: *)(?:[^\s`~]*)\n(?:[\s\S]*?)|`)` checks whether the Wikilink is preceded by the start of a codeblock. The regex for this identification is almost identitical to the one showdown uses as you can see [here](https://github.com/showdownjs/showdown/blob/4af13e5e762f1a4c187cf42f1eced169521dd609/src/subParsers/makehtml/githubCodeBlocks.js#L23). I can make this identical to showdown. It also checks whether the `[[` of a Wikilinks is preceded by a `\``.  
If the Wikilink is preceded by a start of a codeblock or a code span, it does not return a match, otherwise it does (`(?<!` is a negative lookbehind).

Reply:
I think it will be better to simplify this regex massively and not account for wikilinks embedded in codeblocks at all for now. 

The current solution doesn't work reliably enough.

Reply:
Good points, I've addressed this (see below).